### ISSUE TITLE
PG::DatatypeMismatch: ERROR after upgrading to Rails 5

### ISSUE THREAD

**Issue by rjrobinson** [zdennis/activerecord-import#389]
Hey team. Started noticing some odd behavior. 

Basically it goes like this. 

book = Book.new(ebook: 1, author: "William Random")

when I run 
```ruby
Book.import [book], on_duplicate_key: {conflict_target: [:name], columns: [:last_known_source]}
```

And the output is 

```shell
PG::DatatypeMismatch: ERROR:  column "ebook" is of type boolean but expression is of type integer
LINE 1: 1, "William Random"
```
Actions taken. 

If i `book.save` ActiveRecord has no issue placing it in the DB

if I manually cast the 1 to a `true` , active-record & the Importer saves. 

This is new behavior. I haven't touched this code, the only change was that I upgraded to Rails 5, and Ruby 2.3.1

also using PG-9.5.x

**Comment by rjrobinson** [comment]
I think the error comes down to this as a 1 integer or a 0 integer will not be casted as True of False

/import.rb
```ruby
...
array_of_attributes = models.map do |model|
          column_names.map do |name|
            binding.pry
            if stored_attrs.any? && stored_attrs.key?(name.to_sym)
              model.read_attribute(name.to_s)
            else
              model.read_attribute_before_type_cast(name.to_s)
            end
          end
        end

...
```

**Comment by jkowens** [comment]
@rjrobinson thanks for reporting this issue. I think this might be an ActiveRecord 5 regression. I've created a PR at https://github.com/rails/rails/pull/28483 and I'm waiting for feedback there.

**Comment by jkowens** [comment]
BTW the code the code that handles the type casting is here:

https://github.com/zdennis/activerecord-import/blob/master/lib/activerecord-import/import.rb#L684

```
if defined?(type_caster_memo) && type_caster_memo.respond_to?(:type_cast_for_database) # Rails 5.0 and higher
  connection_memo.quote(type_caster_memo.type_cast_for_database(column.name, val))
elsif column.respond_to?(:type_cast_from_user)                                         # Rails 4.2 and higher
  connection_memo.quote(column.type_cast_from_user(val), column)
else                                                                                   # Rails 3.2, 4.0 and 4.1
  if serialized_attributes.include?(column.name)
    val = serialized_attributes[column.name].dump(val)
  end
  connection_memo.quote(column.type_cast(val), column)
end
```

**Comment by jkowens** [comment]
A fix has been included in the v0.18.0 release.
